docs/content/Coding Milestones/PoC2023q1/commands.md

```diff
@@ -227,19 +239,82 @@ objects that enable the syncer to propagate reported state for
 downsynced objects defined by the APIExport from that workspace of a
 subset of the Kubernetes API for managing containerized workloads.
 
-### KubeStellar start
+#### KubeStellar start
 
 This subcommand is used after installation or process stops.
 
 This subcommand stops any running kubestellar controllers and then
 starts them all.  It also does the same thing as `kubestellar init`.
 
-### KubeStellar stop
+#### KubeStellar stop
 
 This subcommand undoes the primary function of `kubestellar start`,
 stopping any running KubeStellar controllers.  It does _not_ tear down
 the ESPW.
 
+## Deployment into a Kubernetes cluster
+
+These commands administer a deployment of the central components ---
+the kcp server, PKI, and the central KubeStellar components --- in a
+Kubernetes cluster that will be referred to as "the hosting cluster".
+
+### Deploy to cluster
+
+Deployment is done with a "kubectl plugin" that creates a [Helm
+"release"](https://helm.sh/docs/intro/using_helm/#three-big-concepts)
+in the hosting cluster. As such, it relies on explicit (on the command
+line) or implicit (in environment variables and/or `~/.kube/config`
+configuration needed to execute Helm commands). An invocation of this
+plugin starts with `kubectl kubestellar deploy` and continues with the
+following, in any order.
+
+- `--openshift $bool`, saying whether the hosting cluster is an
+  OpenShift cluster.  If so then a Route will be created to the kcp
+  server; otherwise, an Ingress object will direct incoming TLS
+  connections to the kcp server.  The default is `false`.
+- `--external-endpoint $domain_name:$port`, saying how the kcp server
+  will be reached from outside the cluster.  The given domain name
+  must be something that the external clients will resolve to an IP
+  address where the cluster's Ingress controller or OpenShift router
+  will be listening, and the given port must be the corresponding TCP
+  port number.  The default value is `localhost:6443`, but this is
+  unlikely to be what you want.
+- a command line flag for the `helm upgrade` command.
+
+For example, to deploy to a plain Kubernetes cluster whose Ingress
+controller can be reached at
+`my-long-application-name.my-region.some.cloud.com:8443`, you would
+issue the following command.
+
+```shell
+kubectl kubestellar deploy --external-endpoint my-long-application-name.my-region.some.cloud.com:8443
+```
+
+### Fetch kubeconfig for internal clients
```

Comment:
Today there are only two kubeconfigs, one for use by clients inside the cluster and one for use by clients that can reach the cluster by its external hostname (which includes the clients that can get there by a longer network path than going direct to the server pod).  Today the kubeconfig for internal clients uses the IP address that the one and only KubeStellar Pod had during startup, which will continue working unless and until that Pod is deleted and replaced.  In the future we plan to do better, using a Service's DNS name rather than an IP address.  I plan to do that in a follow-on PR.